### ISSUE TITLE
Engine callers should be looking at wait events instead of waits on session

### ISSUE THREAD

**Issue by rowanseymour** [nyaruka/goflow#614]
Currently `node > router > wait` is a part of the flow definition.. but also an object that we add to the session blob and set a `timeout_on` property on which is not part of the flow definition. Now that we're adding "timeout_category_uuid" to the definition part of a wait, it's starting to feel a bit weird copying a part of the definition to the session and I'm wondering if these aren't two different things, i.e.

wait in definition..
 * type
 * hint
 * relative timeout (int)
 * router category

wait in session
 * type 
 * hint
 * absolute timeout (datetime)

but for the latter we already have something that communicates that information to the caller.. the `msg_wait` event... so maybe callers that look at waits (simulator, surveyor, mailroom), could just handle the wait via that event?

Furthermore the engine itself looks at the wait on the node we're resuming from and not the wait on the session because the flow may have been changed and no longer have a wait on that node.

**Comment by nicpottier** [comment]
Agree they are different things. Suppose looking through events is not a whole lot different than us looking through events to send messages etc, so maybe that's just fine.

**Comment by rowanseymour** [comment]
Have a bit of a chicken and egg situation. `msg_wait` was missing `hint`.. so I can't go and convert the simulator to look at `msg_wait` events instead of `session.wait`.. until we've updated goflow.. but that's probably not going to happen til this refactor is complete. So I'll put something on session that looks like the current wait to the simulator... and finish this refactor.
